pkg/sbom/nodelist.go

```diff
@@ -266,6 +266,22 @@ func copyEdgeList(original []*Edge) []*Edge {
 	return nodeCopy
 }
 
+// Copy returns a duplicate of the NodeList.
+func (nl *NodeList) Copy() *NodeList {
+	nlo := &NodeList{}
+
+	for _, n := range nl.Nodes {
+		nlo.Nodes = append(nlo.Nodes, n.Copy())
+	}
+	for _, e := range nl.Edges {
+		nlo.Edges = append(nlo.Edges, e.Copy())
+	}
```

Comment:
Not going to block on this, but for reference, we are building utility functions to handle these centrally:

```suggestion
        nlo.Edges = copyEdgeList(nl.Edges),
```